prover/src/constraints/evaluator.rs

```diff
@@ -83,10 +83,17 @@ impl<'a, A: Air, E: FieldElement<BaseField = A::BaseField>> ConstraintEvaluator<
             "extended trace length is not consistent with evaluation domain"
         );
 
-        // build a list of constraint divisors; currently, all transition constraints have the same
-        // divisor which we put at the front of the list; boundary constraint divisors are appended
-        // after that
-        let mut divisors = vec![self.transition_constraints.divisor().clone()];
+        // Build a list of constraint divisors; all transition constraints have the same
+        // divisor which we put at the front of the list; any custom divisor is considered during evaluation
+        // of the constraints by multiplying with default_divisor/custom_divisor each constraint.
+        // This avoids doing unecessary inversions per divisor.
+        // Currently this only consider the default divisor and the boundary constraints divisors
+        let mut divisors = vec![self
+            .transition_constraints
+            .divisors()
+            .first()
+            .unwrap()
+            .clone()];
```

Comment:
```suggestion
        let mut divisors = vec![self.transition_constraints.divisors()[0].clone()];
```
I would directly access the first element that way, because `first()` is typically used in cases where we want to handle the empty vec case (returning None) while here we're unwrapping anyway, so it's cleaner to directly access the first element through `[0]`.

Reply:
This is also what's being used just below.